app/services/katello/pulp3/content.rb

```diff
@@ -4,12 +4,18 @@ module Pulp3
     class Content
       extend Katello::Abstract::Pulp::Content
       class << self
-        def create_upload(size = 0, checksum = nil, content_type = nil)
+        def create_upload(size = 0, checksum = nil, content_type = nil, generic_content_type = nil, repository = nil)
           content_unit_href = nil
           if checksum
             content_backend_service = SmartProxy.pulp_primary.content_service(content_type)
-            content_list = content_backend_service.content_api.list("sha256": checksum)
-            content_unit_href = content_list.results.first.pulp_href unless content_list.results.empty?
+            if repository&.generic?
+              # pulp python doesn't support filter by sha256 (i.e. list method), so need to get the result differently
```

Comment:
I'm surprised the filtering doesn't work since each package does have a sha256 hash.  I think we should make a feature request: https://github.com/pulp/pulp_python/issues

Reply:
It looks like they added filtering for sha256 after I made the first issue. I guess this code will have to be updated whenever the next version of pulp python is released/supported.

Reply:
Let's make an issue for it after this PR merges.

Reply:
i mentioned this above, but i think we should just go ahead and use the new feature before merging this since this won't always work, open to other thoughts though